java_tests/src/test/java/org/lsst/sal/BaseTestCase.java

```diff
@@ -54,15 +52,25 @@ void sleep(long sleep) {
      *                 or SAL__NO_UPDATES.
      * @param data     Data struct to fill.
      */
-    void getTopic(final String funcName, Object data) throws NoSuchMethodException, IllegalAccessException,
-            InvocationTargetException {
+    void getTopic(final String funcName, Object data) {
         Class<?> cls = this.remote.getClass();
-        final Method m = cls.getMethod(funcName, data.getClass());
+        final Method m;
+        try {
+            m = cls.getMethod(funcName, data.getClass());
+        } catch (NoSuchMethodException e) {
+            Assert.fail("The method " + funcName + " should exist.");
+            return;
+        }
         final long startTime = System.currentTimeMillis();
-        Integer retcode = SAL_Test.SAL__NO_UPDATES;
+        int retcode = SAL_Test.SAL__NO_UPDATES;
         while (System.currentTimeMillis() - startTime < TIME_TO_WAIT) {
-            retcode = (Integer) m.invoke(remote, data);
-            sleep(STD_SLEEP);
+            try {
+                retcode = (Integer) m.invoke(remote, data);
+            } catch (IllegalAccessException e) {
+                Assert.fail("The invoked method " + funcName + " should be public.");
+            } catch (InvocationTargetException e) {
+                Assert.fail("The method " + funcName + " should exist.");
```

Comment:
How can this to be raised when you already test for the method to exist at line 58? Please consider adding a code comment explaining.

Reply:
As I explained before, Java has the annoying concept of checked exceptions which HAVE to be handled, even if you are sure that no exception will be thrown. It is part of the API and there is no way around it. This is an integral part of the Java programming language and any Java developer knows this.